### ISSUE TITLE
[FR] Support using a similarity threshold in Firestore retriever

### ISSUE THREAD

**Issue by ssbushi** [firebase/genkit#653]
**Is your feature request related to a problem? Please describe.**
Firestore retriever does not support using a similarity threshold (instead of `k`) when retrieving documents.

**Describe the solution you'd like**
An optional `threshold` config that lets developers specify a threshold for the retrieved documents.

**Describe alternatives you've considered**
Existing workaround:
 - Manually implement a cutoff mechanism as a post-retrieval step.

Consider extending this support to other retrievers.

**Additional context**
[See Discord](https://discord.com/channels/1255578482214305893/1256607873379860563)

**Comment by ssbushi** [comment]
@mbleigh Yes, this is not natively supported in Firestore. Which means that this should be handled in the retriever definition after Firestore returns the appropriate documents.

**Comment by MarkDuckworth** [comment]
This is a feature we are considering implementing natively.

**Comment by ssbushi** [comment]
This is now natively supported in Firestore https://firebase.google.com/docs/firestore/vector-search#distance-threshold

**Comment by ssbushi** [comment]
Fixed in https://github.com/firebase/genkit/pull/2246
